### DISCUSSION TITLE
How to install MEEP using conda for python 3.12 to work with gdsfactory

### DISCUSSION THREAD

**Discussion by Argonghost** [NanoComp/meep#2872] [Q&A]
Hi,

I have a linux subsystem for Windows.  and im trying to install MEEP in order to work with gdsfactory.  I currently have ``gdsfactory`` with Python 3.12 which works.  When i followed the steps to download MEEP using Conda, it creates a virtual env thats Python 3.8.  I need MEEP to be installed on Python 3.12.  How do i do that?
Below are the steps i followed to install MEEP

``wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
 bash miniconda.sh -b -p miniconda3
export PATH=miniconda3/bin:$PATH 
~/miniconda3/bin/conda init bash
conda create -n meep -c conda-forge pymeep pymeep-extras 
source activate meep``

This launches a virtual env thats Python 3.8, which gives many errors with gdsfactory.  Also, whatever i try to do to reinstll gdsfactory using ``pip`` doesnt get reflected .

Thanks

**Comment by gummyequation** [comment]
Open a terminal and activate the environment you want to update.
` conda activate meep `
then tell conda  to install python 3.11.9 in the current environment 
` conda install python=3.11.9`
then voila. 
From what I  remember this is the latest version of python that _conda_ currently supports. To use any later version requires a lot of work from what I remember.
To check which interpreter your environment is using, type the following
```
import sys
print(sys.executable)
```
 If you're working with conda  then the pip commands only works through ipython (at least it works for me) and that's because I use my conda environment as the python interpreter for my IDE (I use spyder).
If you are making changes to python through pip or through conda that are not registering that means whatever python you're running is not the one you're updating. In other words, you are probably using the system's default interpreter instead that of the environment.

**Comment by Argonghost** [comment]
> Open a terminal and activate the environment you want to update.
> 
> ` conda activate meep `
> 
> then tell conda  to install python 3.11.9 in the current environment 
> 
> ` conda install python=3.11.9`
> 
> then voila. 
> 
> From what I  remember this is the latest version of python that _conda_ currently supports. To use any later version requires a lot of work from what I remember.
> 
> To check which interpreter your environment is using, type the following
> 
> ```
> 
> import sys
> 
> print(sys.executable)
> 
> ```
> 
>  If you're working with conda  then the pip commands only works through ipython (at least it works for me) and that's because I use my conda environment as the python interpreter for my IDE (I use spyder).
> 
> If you are making changes to python through pip or through conda that are not registering that means whatever python you're running is not the one you're updating. In other words, you are probably using the system's default interpreter instead that of the environment.
> 
> 

Perfect!! 
Worked like a gem.  Yes indeed it was an issue of pip packages not being reflected in Conda (I use VSCode) 

Thanks for the save!

**Reply by reemastha706** [reply]
Hi, I followed the same, but I get 
`Channels:
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package pymeep-1.29.0-nompi_py38h1234567_103 requires python_abi 3.8.* *_cp38, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ pymeep =* * is installable and it requires
│  └─ python_abi =3.8 *_cp38, which requires
│     └─ python =3.8 *_cpython, which can be installed;
└─ python =3.11.9 * is not installable because it conflicts with any installable versions previously reported.
`
Did you just follow the same?
